src/components/empty-search/index.tsx

```diff
@@ -0,0 +1,66 @@
+import { motion } from 'framer-motion';
+import { PropsWithChildren } from 'react';
+import { Image } from '@atb/components/icon';
+
+import style from './empty-search.module.css';
+import { Typo } from '../typography';
+import { ComponentText, useTranslation } from '@atb/translations';
+import React from 'react';
+
+export type EmptyLoadingSearchProps = PropsWithChildren<{
+  isSearching: boolean;
+  type: keyof typeof ComponentText.EmptySearch.searching;
+}>;
+
+export default function EmptyLoadingSearch({
+  isSearching,
+  type,
+  children,
+}: EmptyLoadingSearchProps) {
+  const { t } = useTranslation();
+
+  const hasEmptyChild = React.Children.toArray(children).some((child) => {
+    if (React.isValidElement(child)) {
+      return 'empty' in child.props;
+    }
+    return false;
+  });
+
+  return (
+    <>
+      {!isSearching && hasEmptyChild && (
```

Comment:
Would it be clean to bake this into a switch?

Reply:
I think when extending to error / geolocation we should bake this into type system